### ISSUE TITLE
Druid UI (datasources pages) crashes if all segments are not loaded.

### ISSUE THREAD

**Issue by amal-v** [apache/druid#10558]
### Description
While the segments are unavailable(loading), the UI tries to populate the datasource pages with this query and fails.
```
2020-11-04T13:37:55,948 WARN [sql[ae2011e4-8731-4b9a-beaf-81513ffec9c9]] org.apache.druid.sql.http.SqlResource - Failed to handle query: SqlQuery{query='SELECT
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   datasource,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   COUNT(*) FILTER (WHERE (is_published = 1 AND is_overshadowed = 0) OR is_realtime = 1) AS num_segments,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   COUNT(*) FILTER (WHERE is_available = 1 AND ((is_published = 1 AND is_overshadowed = 0) OR is_realtime = 1)) AS num_available_segments,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   COUNT(*) FILTER (WHERE is_published = 1 AND is_overshadowed = 0 AND is_available = 0) AS num_segments_to_load,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   COUNT(*) FILTER (WHERE is_available = 1 AND NOT ((is_published = 1 AND is_overshadowed = 0) OR is_realtime = 1)) AS num_segments_to_drop,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   SUM("size") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) AS total_data_size,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   SUM("size" * "num_replicas") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) AS replicated_size,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   MIN("size") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) AS min_segment_size,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   (
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:     SUM("size") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) /
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:     COUNT(*) FILTER (WHERE (is_published = 1 AND is_overshadowed = 0))
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   ) AS avg_segment_size,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   MAX("size") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) AS max_segment_size,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   SUM("num_rows") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0) OR is_realtime = 1) AS total_rows,
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   (
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:     SUM("size") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0)) /
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:     SUM("num_rows") FILTER (WHERE (is_published = 1 AND is_overshadowed = 0))
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]:   ) AS avg_row_size
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]: FROM sys.segments
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]: GROUP BY 1', resultFormat=OBJECT, header=false, context={}, parameters=[]}
Nov 04 13:37:55 ip-10-1-130-103 druid-node.sh[1986]: java.lang.ArithmeticException: / by zero
```
The issue resolves ones all the segments are loaded.

**Comment by suneet-s** [comment]
@amal-v Which version of Druid are you using? I've seen the same symptoms that were fixed in #10482 which is released in 0.20.0

**Comment by amal-v** [comment]
I am facing this issue with `.20`.
![image](https://user-images.githubusercontent.com/15118197/98189829-c8f59000-1f3b-11eb-9020-254efaed3c75.png)
![image](https://user-images.githubusercontent.com/15118197/98189849-d27ef800-1f3b-11eb-9ac5-5354ea9fc964.png)

**Comment by suneet-s** [comment]
@amal-v I found the correct issue where this was fixed - https://github.com/apache/druid/pull/10496/files#diff-b424606a1aed0c701cf23b70d252bab3b445c334a46fd4f5909c7ced137816ccR230-R244

It did not make the cut for 0.20.0, but will be fixed in the next release

**Comment by suneet-s** [comment]
Fixed by #10496
